lib/ex_doc/formatter/html/autolink.ex

```diff
@@ -22,20 +22,20 @@ defmodule ExDoc.Formatter.HTML.Autolink do
   defp all_docs(module, modules) do
     locals = Enum.map module.docs, &(doc_prefix(&1) <> &1.id)
 
-    if moduledoc = module.moduledoc do
```

Comment:
Yes, I know that the syntax is valid :+1:

I just feel (please, correct me if I'm wrong) that this kind of tricks at some point could lead a bad practice, I don't expect to do a *match* inside of a *conditional statement* (the original goal), it's like I'm trying to accomplish more than one thing in just one statement. Also, it could be possible that somebody assume something like this: *Ah, this guy actually wants to do a `==` and after that change :boom:*, or what happens in case of a bad match? I know this not the case, but just thinking about another possible scenarios.